x/concentrated-liquidity/incentives.go

```diff
@@ -970,8 +975,16 @@ func (k Keeper) claimAllIncentivesForPosition(ctx sdk.Context, positionId uint64
 
 				var forfeitedIncentivesPerShare sdk.DecCoins
 				for _, coin := range collectedIncentivesForUptime {
-					// updated forfeitedIncentivesPerShare to add back = collectedIncentivesPerShare / totalSharesAccum
-					forfeitedIncentivesPerShare = append(forfeitedIncentivesPerShare, sdk.NewDecCoinFromDec(coin.Denom, coin.Amount.ToDec().Add(dust.AmountOf(coin.Denom)).Quo(totalSharesAccum)))
+					// updated forfeitedIncentivesPerShare to add back = collectedIncentivesPerShare / (totalSharesAccum - position.NumShares)
+					denominator := totalSharesAccum.Sub(pos.NumShares)
+					// If this is the last position in the accumulator, then the denominator will be zero.
+					// We instead reinvest the forfeited incentives back into the accumulator, because the contract here is that this method will
+					// be called a second time when withdrawing the final position, but during the second time, the totalSharesAccum will be zero so
+					// those incentives will be sent to the community pool in the logic directly above this.
```

Comment:
Why is this called twice?

Reply:
It is called at least once every time we withdraw a position here https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/lp.go#LL202-L205C3

Then, if the position is the last position, we call it a second time here https://github.com/osmosis-labs/osmosis/blob/main/x/concentrated-liquidity/lp.go#LL237-L239C4

The second time we call it though, the totallAccumValue is zero, so we send the remaining amt to the comm pool

Reply:
I see, thank you. I don't think this is in scope for this PR but a general impression I have is that it would be much cleaner if we could special-case claiming for the very last position by a separate `claimAllIncentivesForLastPosition` method.

It feels to me like we keep introducing hack after hack when the solution is to have a separate method for claiming the last one.

Reply:
Couldn't agree more with this solution